scripts/make_source_dist.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash -ex
+# vim: ts=4 sw=4 expandtab
+ceph_version_git=$(git describe --abbrev=8 --match 'v*' | sed s/^v//)
+
+if command -v podman; then
+  PODMAN=podman
+elif [[ "$(groups)" =~ .*\ docker\ .* ]]; then
+  PODMAN=docker
+else
+  PODMAN="sudo docker"
+fi
+
+printf "FROM ubuntu:24.04\nRUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y dpkg-dev devscripts && apt-get clean && rm -rf /var/lib/apt/lists/*" | $PODMAN build -t ubuntu_builder -
+$PODMAN run --rm -v $PWD:/ceph:z -e DEBEMAIL=contact@ceph.com ubuntu_builder:latest bash -c "cd /ceph && dch -v $ceph_version_git-1 'autobuilder' < /dev/null"
+
+rm -rf dist release
+mkdir -p release
+echo $ceph_version_git > release/version
+
+# declare an associative array to map file extensions to tar flags
+declare -A compression=( ["bz2"]="j" ["gz"]="z" ["xz"]="J" )
+for cmp in "${!compression[@]}"; do
+  rm -f ceph-*.tar.$cmp
+done
+echo building tarball
+./make-dist $ceph_version_git
+for cmp in "${!compression[@]}"; do
+  extension="tar.$cmp"
+  ceph_version_tarball=$(ls ceph-*.$extension | cut -c 6- | sed "s/.$extension//" || true)
+  flag="${compression[$cmp]}"
+  extract_flags="${flag}xf"
+  compress_flags="${flag}cf"
+  if [ "$ceph_version_tarball" != "" ]; then break; fi
+done
+echo tarball vers $ceph_version_tarball
+
+echo extracting
+mkdir -p release/$ceph_version_git
+cd release/$ceph_version_git
+
+# FIXME: is the rpm patch stuff below ever necessary?
+# mkdir -p release/$ceph_version_git/rpm
+# cp rpm/*.patch release/$ceph_version_git/rpm || true
+
+tar $extract_flags $WORKSPACE/ceph/ceph-$ceph_version_tarball.$extension
+
+[ "$ceph_version_tarball" != "$ceph_version_git" ] && mv ceph-$ceph_version_tarball ceph-$ceph_version_git
+
+tar zcf ceph_$ceph_version_git.orig.tar.gz ceph-$ceph_version_git
+# FIXME delete this if it's not needed
+ln ceph_$ceph_version_git.orig.tar.gz ceph-$ceph_version_git.tar.gz
```

Comment:
I believe Debian packaging uses orig.tar.gz and the specfile uses tar.gz, although I thought that the Debian orig.tar.gz was not supposed to include the debian/ dir, and all the references I can find say that, but, ours does and apparently always has.  I guess maybe it's a "native" package and not a "non-native" package, and so it's assumed it has debian/ included.  in which case naming it 'orig' makes little sense.  I don't *(&#$ know.

Reply:
Turns out we are only using the bz2! https://jenkins.ceph.com/job/ceph-dev-pipeline/288/pipeline-console/?selected-node=234